eclair-core/src/test/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreWalletSpec.scala

```diff
@@ -68,6 +68,105 @@ class BitcoinCoreWalletSpec extends TestKitBaseClass with BitcoindService with A
+    val txids = locks.map { item =>
+      val JString(txid) = item \ "txid"
+      val JInt(vout) = item \ "vout"
+      OutPoint(ByteVector32.fromValidHex(txid).reverse, vout.toInt)
+    }
+    txids.toSet
+  }
+
+  test("unlock transaction inputs if publishing fails") {
+    val bitcoinClient = new BasicBitcoinJsonRPCClient(
+      user = config.getString("bitcoind.rpcuser"),
+      password = config.getString("bitcoind.rpcpassword"),
+      host = config.getString("bitcoind.host"),
+      port = config.getInt("bitcoind.rpcport"))
+    val wallet = new BitcoinCoreWallet(bitcoinClient)
+
+    val sender = TestProbe()
+    val pubkeyScript = Script.write(Script.pay2wsh(Scripts.multiSig2of2(randomKey.publicKey, randomKey.publicKey)))
+
+    // create a huge tx so we make sure it has > 1 inputs
+    wallet.makeFundingTx(pubkeyScript, Btc(250), 1000).pipeTo(sender.ref)
+    val MakeFundingTxResponse(fundingTx, outputIndex, _) = sender.expectMsgType[MakeFundingTxResponse]
+
+    // spend the first 2 inputs
+    val tx1 = fundingTx.copy(
+      txIn = fundingTx.txIn.take(2),
+      txOut = fundingTx.txOut.updated(outputIndex, fundingTx.txOut(outputIndex).copy(amount = Btc(50)))
+    )
+    wallet.signTransaction(tx1).pipeTo(sender.ref)
+    val SignTransactionResponse(tx2, true) = sender.expectMsgType[SignTransactionResponse]
+
+    wallet.commit(tx2).pipeTo(sender.ref)
+    assert(sender.expectMsgType[Boolean])
+
+    // fundingTx inputs are still locked except for the first 2 that were just spent
+    val expectedLocks = fundingTx.txIn.drop(2).map(_.outPoint).toSet
+    awaitCond({
+      val locks = getLocks(sender)
+      expectedLocks -- locks isEmpty
+    }, max = 10 seconds, interval = 1 second)
+
+    // publishing fundingTx will fail as its first 2 inputs are already spent by tx above in the mempool
+    wallet.commit(fundingTx).pipeTo(sender.ref)
+    val result = sender.expectMsgType[Boolean]
+    assert(!result)
+
+    // and all locked inputs should now be unlocked
+    awaitCond({
+      val locks = getLocks(sender)
+      locks isEmpty
+    }, max = 10 seconds, interval = 1 second)
+  }
+
+  test("unlock outpoints correcly") {
+    val bitcoinClient = new BasicBitcoinJsonRPCClient(
+      user = config.getString("bitcoind.rpcuser"),
+      password = config.getString("bitcoind.rpcpassword"),
+      host = config.getString("bitcoind.host"),
+      port = config.getInt("bitcoind.rpcport"))
+
+    val wallet = new BitcoinCoreWallet(bitcoinClient)
+
+    val sender = TestProbe()
+    val pubkeyScript = Script.write(Script.pay2wsh(Scripts.multiSig2of2(randomKey.publicKey, randomKey.publicKey)))
+
+    {
+      // test #1: unlock outpoints that are actually locked
+      // create a huge tx so we make sure it has > 1 inputs
+      wallet.makeFundingTx(pubkeyScript, Btc(250), 1000).pipeTo(sender.ref)
+      val MakeFundingTxResponse(fundingTx, outputIndex, _) = sender.expectMsgType[MakeFundingTxResponse]
```

Comment:
Why not add `assert`s on the number of inputs since the test depends on that.
```suggestion
      // create a huge tx so we make sure it has > 1 inputs
      wallet.makeFundingTx(pubkeyScript, Btc(250), 1000).pipeTo(sender.ref)
      val MakeFundingTxResponse(fundingTx, outputIndex, _) = sender.expectMsgType[MakeFundingTxResponse]
      assert(fundingTx.txIn.size > 2)
```

Reply:
I thought it was not needed since here our wallet only has 50 btc coinbase txs and needs 6 of them to create a 250 btc tx +change but ok